### PR TITLE
Improve HDF icon resolution and macOS

### PR DESCRIPTION
**Pull request by oyarzun**
@byrnHDF  This is an attempt to improve the HDF icon resolution by allow SWT to select the appropriate icon via `setImages`. 

It adds a macOS styled HDF icon.

Screen Shot 2022-09-10 at 5 21 51 PM

It also does not change the icon in the macOS Dock when opening a Dataset.

Screen Shot 2022-09-10 at 5 37 27 PM

### CODE REVIEW

File: src/org.hdfgroup.hdfview/hdf/view/TableView/DefaultScalarDSTableView.java [hunk 1]
```diff
@@ -113,7 +113,12 @@ public DefaultScalarDSTableView(DataViewManager theView, HashMap dataPropertiesM
         super(theView, dataPropertiesMap);
 
         if (!shell.isDisposed()) {
-            shell.setImage(dataObject.getDatatype().isText() ? ViewProperties.getTextIcon() : ViewProperties.getDatasetIcon());
+            if (System.getProperty("os.name").toLowerCase().contains("mac")) {
+                shell.setImages(ViewProperties.getHdfIcons());
```
Comment (jhendersonHDF) [inline src/org.hdfgroup.hdfview/hdf/view/TableView/DefaultScalarDSTableView.java:117]: I can see from your example screenshots that the dataset shell still uses the Dataset icon, but wouldn't this line explicitly set its icon to the HDF icon on macOS?

Comment (oyarzun) [reply src/org.hdfgroup.hdfview/hdf/view/TableView/DefaultScalarDSTableView.java:117]: @jhendersonHDF yes, it keeps dock icon as the HDF icon when you open a dataset (i.e. the second icon from the right). The icon on the right side is the original code using HDFView 3.1.4 downloaded from HDFGroup's website.

I just seems odd for an application to change the macOS dock icon.

Comment (jhendersonHDF) [reply src/org.hdfgroup.hdfview/hdf/view/TableView/DefaultScalarDSTableView.java:117]: Ah ok. Generally on Windows and Linux each dataset would be opened in a separate window with its own Dataset icon, while the main HDFView window would keep the HDF icon; I guess that isn't the case for macOS? In any case I think this is a great improvement for users of HDFView on macOS.

Comment (oyarzun) [reply src/org.hdfgroup.hdfview/hdf/view/TableView/DefaultScalarDSTableView.java:117]: They still get open in a new window on macOS, but there is no icon in the corner of the window like there is in Windows and Linux.

Screen Shot 2022-09-12 at 12 45 56 PM

**Comment by byrnHDF** [comment]
The PR should go against the master branch first, and then be committed to the 3.2 and 3.1 branches. I am doing some major refactoring of the code right now, so I wouldn't worry about the other branches right now. Let's merge to master and I'll grab the changes.

**Comment by oyarzun** [comment]
Closing in favor of new PR #76 to merge into master

**Pull request closed without merging**